File: app/controllers/api/language.py
from services.languages import languages


from flask import Blueprint, request, jsonify


api_languages = Blueprint('api/language/', __name__)


@api_languages.route("/<language_type>/execute", methods=["POST"])
def execute_language(language_type):
    result = languages.process(language_type, request.json["code"], request.json["user_id"], request.json["file_id"])
    return jsonify(result)

File: app/static/js/codeContainer/data/language.js
export async function updateFileAndRunCode(languageType, code, userId, fileId) {
    const request = await fetch(`/api/language/${languageType}/update-file-and-run-code`, {
        method: "POST",
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ code, userId, fileId })
    })
    return await request.json()
}

export async function runCode(languageType, code) {
    const request = await fetch(`/api/language/${languageType}/run-code`, {
        method: "POST",
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ code })
    })
    return await request.json()
}
